### PR TITLE
Case sensative bug

### PR DESCRIPTION
**Pull request by dpwspoon**
Ready to merge

### CODE REVIEW

**Comment by dpwspoon** [comment]
See commits, one did white spacing fix

File: server/src/test/java/org/kaazing/gateway/server/config/parse/ServiceDefaultsTest.java [hunk 1]
```diff
@@ -69,8 +69,8 @@ public void testDefaultConnectOptions() throws Exception {
         Assert.assertTrue("TLSv1".equals(sslProtocols[0]));
 
         Assert.assertTrue("en0".equals(connectOptionsMap.get("udp.interface")));
-        System.out.println(connectOptionsMap.get("tcp.transport"));
-        Assert.assertTrue("socks://localhost:8000".equals(connectOptionsMap.get("tcp.transport").toString().trim()));
+        final String trim = connectOptionsMap.get("tcp.transport").toString().trim();
```
Comment (jitsni) [inline server/src/test/java/org/kaazing/gateway/server/config/parse/ServiceDefaultsTest.java]: rename trim to tcpTransport, also remove final (I feel final for local variables is just noise).

File: resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java [hunk 2]
```diff
@@ -134,7 +135,7 @@ public int compare(ResourceAddress addr1, ResourceAddress addr2) {
             String authority1 = location1.getAuthority();
             String authority2 = location2.getAuthority();
             
-            int compareAuthority = STRING_COMPARATOR.compare(authority1, authority2);
+            int compareAuthority = CASE_INSENSITIVE_ORDER.compare(authority1, authority2);
```
Comment (jitsni) [inline resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java]: Do you want to wrap CASE_INSENSIVE_ORDER in compareComparable ? STRING_COMPARATOR is doing that and it further wraps in compareNonNull (so it protects from null authority values)

File: resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/uri/URIUtils.java [hunk 3]
```diff
@@ -622,5 +622,63 @@ private String buildURIFromTokens(String scheme, String authority, String path,
...
+                try {
+                    canonicalURI = buildURIAsString(scheme, getUserInfo(uri), host == null ?
+                            null : host.toLowerCase(), getPort(uri), path, getQuery(uri), getFragment(uri));
+                } catch (URISyntaxException ex) {
+                    throw new IllegalArgumentException("Invalid URI: " + uri + " in Gateway configuration file", ex);
+                }
+            }
+        }
+        return canonicalURI;
+    }
```
Comment (jitsni) [inline resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/uri/URIUtils.java]: Looks like this code is just getting moved to this file from GatewayContextResolver.

Comment (dpwspoon) [inline resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/uri/URIUtils.java]: I can put it back if you think its better there (or in a different Util class in gateway.server...?  DefaultServiceContext now uses this, and originally I was thinking the ServiceContext / other  stuff in service module needed access to this method, which is why I moved it...

**Comment by jitsni** [comment]
Can you run enterprise.gateway once with this changeset (because comparators change affects things drastically).

**Pull request merged**